### PR TITLE
artifactregistry: fix comments, more examples for common remote repositories

### PR DESCRIPTION
**Pull request by Gridasov**
Common repositories for Artifact Registry can accept the following resources:
- URI to another Artifact Registry instance
- URI to external registries

This PR fixes comments for `google_artifact_registry_repository` and adds corresponding examples. 

```release-note:none

```

### CODE REVIEW

File: mmv1/products/artifactregistry/Repository.yaml [hunk 1]
```diff
@@ -521,7 +545,7 @@ properties:
           - name: 'customRepository'
             type: NestedObject
             description: |-
-              Settings for a remote repository with a custom uri.
+              [Deprecated, please use commonRepository instead] Settings for a remote repository with a custom uri.
```
Comment (yihanzhen) [inline mmv1/products/artifactregistry/Repository.yaml:548]: Just want to make sure, the common_repository are well-tested and stable enough in the probe results that we can safely ask users to use it ove the old configs?

**Review by yihanzhen — APPROVED** [review]

File: mmv1/products/artifactregistry/Repository.yaml (hunk 1 above)
Comment (Gridasov) [reply mmv1/products/artifactregistry/Repository.yaml:548]: They are already used everywhere by UI and gcloud instead of Custom repositories.

**Comment by yihanzhen** [comment]
LGTM

**Review by NickElliot — APPROVED** [review]
LGTM!

**Pull request merged**